From efb826cf7ddb0469ac316b7971a9397aa5c31711 Mon Sep 17 00:00:00 2001
From: Huang Shan <huang.shan@freescale.com>
Date: Tue, 13 May 2014 14:31:20 +0800
Subject: [PATCH 067/115] ENGR00303833-3 ethernet:Restore proxy after reboot
 if set

Why not effect after reboot? Null name is filled in old logic.
Reset proxy after each proxy information is changed.

Signed-off-by: Huang Shan <huang.shan@freescale.com>
---
 Ethernet/com/fsl/ethernet/EthernetAdvDialog.java | 18 +++++++-----------
 Ethernet/com/fsl/ethernet/EthernetManager.java   |  8 ++++++++
 Ethernet/com/fsl/ethernet/MainActivity.java      |  1 +
 3 files changed, 16 insertions(+), 11 deletions(-)

diff --git a/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java b/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
index 8784aff..b4d1323 100644
--- a/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
+++ b/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
@@ -77,17 +77,13 @@ public class EthernetAdvDialog extends AlertDialog implements DialogInterface.On
 
     public void handle_saveconf() {
         EthernetDevInfo info = new EthernetDevInfo();
-        if ((mProxyIp.getText().toString().equals(""))||(mProxyPort.getText().toString().equals("")))
-        {
-        }else{
-            info.setIfName("eth0");
-            info.setConnectMode(mEthEnabler.getManager().getSharedPreMode());
-            info.setIpAddress(mEthEnabler.getManager().getSharedPreIpAddress());
-            info.setDnsAddr(mEthEnabler.getManager().getSharedPreDnsAddress());
-            info.setProxyAddr(mProxyIp.getText().toString());
-            info.setProxyPort(mProxyPort.getText().toString());
-            info.setProxyExclusionList(mProxyExclusionList.getText().toString());
-        }
+	info.setIfName("eth0");
+	info.setConnectMode(mEthEnabler.getManager().getSharedPreMode());
+	info.setIpAddress(mEthEnabler.getManager().getSharedPreIpAddress());
+	info.setDnsAddr(mEthEnabler.getManager().getSharedPreDnsAddress());
+	info.setProxyAddr(mProxyIp.getText().toString());
+	info.setProxyPort(mProxyPort.getText().toString());
+	info.setProxyExclusionList(mProxyExclusionList.getText().toString());
         mEthEnabler.getManager().updateDevInfo(info);
         mEthEnabler.getManager().setProxy();
     }
diff --git a/Ethernet/com/fsl/ethernet/EthernetManager.java b/Ethernet/com/fsl/ethernet/EthernetManager.java
index fac1edd..d21ba9a 100644
--- a/Ethernet/com/fsl/ethernet/EthernetManager.java
+++ b/Ethernet/com/fsl/ethernet/EthernetManager.java
@@ -382,8 +382,16 @@ public class EthernetManager {
         }
         ProxyProperties proxyProperties =
             new ProxyProperties(getSharedPreProxyAddress(), port, exclusionList);
+        mConnMgr.setGlobalProxy(null);
         mConnMgr.setGlobalProxy(proxyProperties);
         Log.i(TAG,"=============getHttpProxy==============" + proxyProperties);
     }
 
+    public void initProxy(){
+        EthernetDevInfo info = getSavedConfig();
+        if (info != null) {
+            updateDevInfo(info);
+            setProxy();
+	}
+    }
 }
diff --git a/Ethernet/com/fsl/ethernet/MainActivity.java b/Ethernet/com/fsl/ethernet/MainActivity.java
index f2abc3b..08cd7d6 100644
--- a/Ethernet/com/fsl/ethernet/MainActivity.java
+++ b/Ethernet/com/fsl/ethernet/MainActivity.java
@@ -58,6 +58,7 @@ public class MainActivity extends Activity {
                 Context.MODE_WORLD_WRITEABLE);
 
         mEthEnabler = new EthernetEnabler(this);
+        mEthEnabler.getManager().initProxy();
         mEthConfigDialog = new EthernetConfigDialog(this, mEthEnabler);
         mEthEnabler.setConfigDialog(mEthConfigDialog);
         mEthAdvancedDialog = new EthernetAdvDialog(this, mEthEnabler);
-- 
1.8.0

